flocker/apiclient/_client.py

```diff
@@ -744,6 +779,21 @@ def list_containers_configuration(self):
         )
         return d
 
+    def list_containers_state(self):
+        d = self._request(b"GET", b"/state/containers", None, {OK})
+
+        def parse(container):
+            return ContainerState(
+                node_uuid=UUID(container['node_uuid']),
```

Comment:
:grey_question: In _parse_configuration_container we explicitly use unicode keys. For consistency, prefer using explicitly unicode keys here as well [u'node_uuid']

Also I take some issue with the client library raising and "AttributeError" if the server is misbehaving (returning a dictionary without a given key). But... I think that is a battle that is already lost. Certainly your code meets the existing standards, so please don't take any action on that grievance.  Mostly I'm just commenting so that you can ease my mind if there actually is some validation of the input that I'm missing (I don't see any input validation in _request_headers, but I know it took me awhile to find it in our server code).